attestation/docker/docker.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2022 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"crypto"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/in-toto/go-witness/cryptoutil"
+	docker "github.com/in-toto/go-witness/internal/docker"
+	"github.com/in-toto/go-witness/log"
+	"github.com/invopop/jsonschema"
+)
+
+const (
+	Name    = "docker"
+	Type    = "https://witness.dev/attestations/docker/v0.1"
+	RunType = attestation.PostProductRunType
+
+	sha256MimeType = "text/sha256+text"
+	jsonMimeType   = "application/json"
+)
+
+// This is a hacky way to create a compile time error in case the attestor
+// doesn't implement the expected interfaces.
+var (
+	_ attestation.Attestor  = &Attestor{}
+	_ attestation.Subjecter = &Attestor{}
+	_ dockerAttestor        = &Attestor{}
+)
+
+type dockerAttestor interface {
```

Comment:
I'm confused why we need this interface. This is just redclaring our other interfaces.

Reply:
This was here simply due to the fact that I copied `attestation/oci/oci.go` and find and replace'd to create the attestor. This interface exists in the case of the `oci` attestor because of the testOCIAttestor in `internal/attestors/oci.go`. Whether we will want a test attestor for this attestor or not, probably a story for another time.